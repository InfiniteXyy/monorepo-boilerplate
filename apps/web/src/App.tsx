import { useState } from 'react';
import { useDeletePost } from './api/useDeletePost';
import { useListPosts } from './api/useListPosts';
import { CreatePost } from './components/CreatePost';
import { PostDetail } from './components/PostDetail';

export function App() {
  const [currentPostId, setCurrentPostId] = useState<string>();
  const { data: posts } = useListPosts();
  const { mutateAsync: deletePost } = useDeletePost();

  return (
    <div className="min-h-screen bg-gray-50">
      <div className="container mx-auto px-4 py-8">
        <div className="grid grid-cols-1 gap-8 md:grid-cols-2">
          <div>
            <CreatePost />
            <div className="mt-8">
              <h2 className="mb-4 text-xl font-semibold">Posts</h2>
              <ul className="space-y-2">
                {posts?.map((post) => (
                  <li
                    key={post.id}
                    onClick={() => setCurrentPostId(post.id)}
                    className={`flex items-center cursor-pointer rounded-lg border p-3 shadow-sm transition-colors ${post.id === currentPostId ? 'bg-blue-100' : 'bg-white hover:bg-blue-50'}`}
                  >
                    {post.title}
                    <button
                      className="h-7 w-7 text-sm rounded-full hover:bg-gray-200 cursor-pointer ml-auto"
                      type="button"
                      aria-label={`Delete ${post.title}`}
                      onClick={async (e) => {
                        e.stopPropagation();
                        await deletePost({ id: post.id });
                        if (currentPostId === post.id) setCurrentPostId(undefined);
                      }}
                    >
                      ‚ùå
                    </button>
                  </li>
                ))}
              </ul>
            </div>
          </div>
          {currentPostId && <PostDetail postId={currentPostId} />}
        </div>
      </div>
    </div>
  );
}
