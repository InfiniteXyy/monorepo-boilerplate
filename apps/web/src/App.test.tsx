import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { App } from './App';

const component = (
  <QueryClientProvider client={new QueryClient()}>
    <App />
  </QueryClientProvider>
);

// Mock the API hook
vi.mock('./api/useListPosts', () => ({
  useListPosts: () => ({
    data: [
      { id: '1', title: 'Test Post 1' },
      { id: '2', title: 'Test Post 2' },
    ],
  }),
}));

describe('app', () => {
  it('renders correctly', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it('renders posts', () => {
    const { getByText } = render(component);
    expect(getByText('Test Post 1')).toBeInTheDocument();
    expect(getByText('Test Post 2')).toBeInTheDocument();
  });
});
