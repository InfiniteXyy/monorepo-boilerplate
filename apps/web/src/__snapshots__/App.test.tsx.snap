// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders correctly 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50"
  >
    <div
      class="container mx-auto px-4 py-8"
    >
      <div
        class="grid grid-cols-1 gap-8 md:grid-cols-2"
      >
        <div>
          <div
            class="rounded-lg border bg-white p-6 shadow-sm"
          >
            <h1
              class="mb-6 text-2xl font-bold"
            >
              Create Post
            </h1>
            <form
              class="space-y-4"
            >
              <div>
                <input
                  class="w-full rounded-lg border p-2 outline-none focus:border-blue-500"
                  name="title"
                  placeholder="Title"
                />
              </div>
              <div>
                <textarea
                  class="w-full rounded-lg border p-2 outline-none focus:border-blue-500"
                  name="description"
                  placeholder="Description"
                  rows="4"
                />
              </div>
              <button
                class="rounded-lg bg-blue-500 px-4 py-2 text-white transition-colors hover:bg-blue-600"
                type="submit"
              >
                Create Post
              </button>
            </form>
          </div>
          <div
            class="mt-8"
          >
            <h2
              class="mb-4 text-xl font-semibold"
            >
              Posts
            </h2>
            <ul
              class="space-y-2"
            >
              <li
                class="cursor-pointer rounded-lg border p-3 shadow-sm transition-colors bg-white hover:bg-blue-50"
              >
                Test Post 1
              </li>
              <li
                class="cursor-pointer rounded-lg border p-3 shadow-sm transition-colors bg-white hover:bg-blue-50"
              >
                Test Post 2
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app > renders correctly 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50"
  >
    <div
      class="container mx-auto px-4 py-8"
    >
      <div
        class="grid grid-cols-1 gap-8 md:grid-cols-2"
      >
        <div>
          <div
            class="rounded-lg border bg-white p-6 shadow-sm"
          >
            <h1
              class="mb-6 text-2xl font-bold"
            >
              Create Post
            </h1>
            <form
              class="space-y-4"
            >
              <div>
                <input
                  class="w-full rounded-lg border p-2 outline-none focus:border-blue-500"
                  name="title"
                  placeholder="Title"
                />
              </div>
              <div>
                <textarea
                  class="w-full rounded-lg border p-2 outline-none focus:border-blue-500"
                  name="description"
                  placeholder="Description"
                  rows="4"
                />
              </div>
              <button
                class="rounded-lg bg-blue-500 px-4 py-2 text-white transition-colors hover:bg-blue-600"
                type="submit"
              >
                Create Post
              </button>
            </form>
          </div>
          <div
            class="mt-8"
          >
            <h2
              class="mb-4 text-xl font-semibold"
            >
              Posts
            </h2>
            <ul
              class="space-y-2"
            >
              <li
                class="cursor-pointer rounded-lg border p-3 shadow-sm transition-colors bg-white hover:bg-blue-50"
              >
                Test Post 1
              </li>
              <li
                class="cursor-pointer rounded-lg border p-3 shadow-sm transition-colors bg-white hover:bg-blue-50"
              >
                Test Post 2
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
